keyboards/crkbd/keymaps/mcrown/oled.c

```diff
@@ -0,0 +1,384 @@
+            break;
+
+        case _QWERTY:
+            oled_write_P(PSTR("Qwerty"), FALSE);
+            break;
+
+        default:
+            oled_write_ln_P(PSTR("Undefined"), FALSE);
+            break;
+    }
+#endif
+}
+
+/** @brief displays the current active layer.
+ *
+ *  If any argument is invalid, the function has no effect.
+ *
+ *  @param void.
+ *  @return void.
+ */
+void render_layer_state(void)
+{
+    bool lower;
+    bool raise;
+    bool adjust;
+    bool numpad;
+
+    lower = layer_state_is(_LOWER) & !layer_state_is(_ADJUST);
+    raise = layer_state_is(_RAISE) & !layer_state_is(_ADJUST);
+    adjust = layer_state_is(_ADJUST);
+    numpad = layer_state_is(_NUMPAD);
+
+#ifdef OLED_VERTICAL
+    oled_write_P(PSTR("\nLyr:\n"), FALSE);
+    if(lower)
+    {
+        oled_write_P(PSTR("Lwr\n"), TRUE);
+    }
+    else if(raise)
+    {
+        oled_write_P(PSTR("Ris\n"), TRUE);
+    }
+    else if(adjust)
+    {
+        oled_write_P(PSTR("Adj\n"), TRUE);
+    }
+    else if(numpad)
+    {
+        oled_write_P(PSTR("Num\n"), TRUE);
+    }
+    else
+    {
+        oled_write_P(PSTR("Def\n"), FALSE);
+    }
+#else
+    oled_write_P(PSTR("\nLayer:"), FALSE);
+    if(lower)
+    {
+        oled_write_P(PSTR(" Lower "), TRUE);
+    }
+    else if(raise)
+    {
+        oled_write_P(PSTR(" Raise "), TRUE);
+    }
+    else if(adjust)
+    {
+        oled_write_P(PSTR(" Adjust "), TRUE);
+    }
+    else if(numpad)
+    {
+        oled_write_P(PSTR(" Numpad "), TRUE);
+    }
+    else
+    {
+        oled_write_P(PSTR(" Default"), FALSE);
+    }
+#endif
+}
+
+/** @brief displays the current status of the main display/
+ *
+ *  If any argument is invalid, the function has no effect.
+ *
+ *  @param void.
+ *  @return void.
+ */
+void render_status(void)
+{
+    render_layout_state();
+    oled_write_P(PSTR("\n"), FALSE);
+    render_layer_state();
+    render_keylogger_status();
+}
+
+/** @brief renders the logo to be displayed.
+ *
+ *  If any argument is invalid, the function has no effect.
+ *
+ *  @param void.
+ *  @return void.
+ */
+static void render_logo(void)
+{
+    static const char PROGMEM qmk_logo[] = {
+        0x80, 0x81, 0x82, 0x83, 0x84, 0x85, 0x86, 0x87, 0x88, 0x89, 0x8A, 0x8B, 0x8C, 0x8D, 0x8E, 0x8F, 0x90, 0x91, 0x92, 0x93, 0x94,
+        0xA0, 0xA1, 0xA2, 0xA3, 0xA4, 0xA5, 0xA6, 0xA7, 0xA8, 0xA9, 0xAA, 0xAB, 0xAC, 0xAD, 0xAE, 0xAF, 0xB0, 0xB1, 0xB2, 0xB3, 0xB4,
+        0xC0, 0xC1, 0xC2, 0xC3, 0xC4, 0xC5, 0xC6, 0xC7, 0xC8, 0xC9, 0xCA, 0xCB, 0xCC, 0xCD, 0xCE, 0xCF, 0xD0, 0xD1, 0xD2, 0xD3, 0xD4,
+        0x00};
+
+    oled_write_P(qmk_logo, FALSE);
+}
+
+/** @brief executes the actions for both displays.
+ *
+ *  If any argument is invalid, the function has no effect.
+ *
+ *  @param void.
+ *  @return void.
+ */
+void oled_task_user(void)
+{
+    if (timer_elapsed32(standby_oled_timer) > 15000)
+    {
+        oled_off();
+    }
+    else
+    {
+        oled_on();
+        if(TRUE==is_master)
```

Comment:
```suggestion
        if(is_keyboard_master())
```

Reply:
I did a research and I thought  ´´´is_master´´´ was an external variable. There are two matches one is a variable while the other is a definition and this is confusing. Being said that I think your suggestion is better. Thanks :)

Reply:
I tried your suggestion but for some reason is not working

Reply:
Are you using the "legacy" version?  If so, that may be why.

Reply:
Yes, legacy is being used here